### PR TITLE
791 add performance data in codegen

### PR DESCRIPTION
**Pull request by fm-117**
Time measurements for Codegen. Example of output :
```
FILE;ITERATION;STEP;TIME_TAKEN
CGM110.cbl;1;CreateTargetDocument;145,6953
CGM110.cbl;1;Qualifier;152,3024
CGM110.cbl;1;Actions;765,8847
CGM110.cbl;1;TreeToCode;5281,6669
CGM110.cbl;2;CreateTargetDocument;154,7135
CGM110.cbl;2;Qualifier;0,0008
CGM110.cbl;2;Actions;133,9764
CGM110.cbl;2;TreeToCode;4751,2268

Time per step :
TreeToCode               (4) : Total =      10032,8937 ms, Avg =      5016,44685 ms
Actions                  (3) : Total =        899,8611 ms, Avg =       449,93055 ms
CreateTargetDocument     (1) : Total =        300,4088 ms, Avg =        150,2044 ms
Qualifier                (2) : Total =        152,3032 ms, Avg =         76,1516 ms

Global Codegen time : Total = 11385,4668 ms, Avg = 2846,3667 ms
```

The first block can be copy-pasted as CSV, second block is more human-readable.

Also fixes bug in LinearGeneration occuring when last lines of a input files are comments.

**Notes**
- Performance tests are Ignored by default
- Test_BigTypes_1Procedure is KO due to a bug (see #1445)
- Test_GlobalStorage is KO due to a bug in CrossChecker but fixed in #1081, see commit aa27f63e17281fed29081ef8a611a871ec4de356

### CODE REVIEW

**Comment by fm-117** [comment]
Still have OutOfMemoryException on CGMV01-UseAlotOfTypes-100Times.tcbl and CGMX02.COB.
Here is a sample of results [codegen_test_sample.xlsx](https://github.com/TypeCobolTeam/TypeCobol/files/3569537/codegen_test_sample.xlsx).

**Comment by maxime645** [comment]
Is it normal that you refered to Issue #791 for your commits ?

File: Codegen/test/Performance.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                    total[pair.Key] += pair.Value;
+                }
+            }
+
+            // Aggregated stats.
+            Console.WriteLine();
+            Console.WriteLine("Time per step :");
+            foreach (var stepData in total.Select((p, i) => new {Index = i + 1, Step = p.Key, Duration = p.Value.TotalMilliseconds}).OrderByDescending(a => a.Duration))
+            {
+                Console.WriteLine($"{stepData.Step,-25}({stepData.Index}) : Total = {stepData.Duration,15} ms, Avg = {stepData.Duration / ITERATION_COUNT,15} ms");
```
Comment (smedilol) [inline Codegen/test/Performance.cs]: Why -25 and 15?

Comment (fm-117) [reply Codegen/test/Performance.cs]: To slightly improve readability of results. -25 left aligns the step name on 25 characters and 15 right aligns the duration. It assumes that the step name is 25 characters long max.

**Comment by fm-117** [comment]
I did not create an issue specifically for the performance tests but issue #791 was the one which would benefit from performance data since the changes made add an extra cost in codegen. So I reused issue #791 as reference.

**Comment by fm-117** [comment]
See #1507

**Pull request closed without merging**